### PR TITLE
[KHRGA-70] Updated DA General accessor concept

### PR DESCRIPTION
**Pull request by PetroBondar**
Issue: KHRGA-70

### CODE REVIEW

File: source/iface/accessor-concept.rst [hunk 1]
```diff
@@ -2,41 +2,400 @@
...
+
+For the image accessors (:ref:`unsampled-image-accessors`,
+:ref:`sampled-image-accessors`), ``DataT`` must be one of:
+
+* ``sycl::int4`` (``sycl::vec<int32_t, 4>``);
+* ``sycl::uint4`` (``sycl::vec<uint32_t, 4>``);
+* ``sycl::float4`` (``sycl::vec<float, 4>``);
+* ``sycl::half4`` (``sycl::vec<half, 4>``).
+
+See :ref:`local_accessor` page for it's allowed ``DataT`` types.
```
Comment (mkinsner) [inline source/iface/accessor-concept.rst:78]:
```suggestion
See :ref:`local_accessor` for it's allowed ``DataT`` types.
```

File: source/iface/accessor-concept.rst [hunk 2]
```diff
@@ -2,41 +2,400 @@
...
+.. _deduction-tags:
+
+==============
+Deduction tags
+==============
+
+Some accessor constructors take a ``TagT`` parameter, which is
+used to deduce template arguments for the constructor's class.
+
+Each of the access modes in table above has an associated tag,
```
Comment (mkinsner) [inline source/iface/accessor-concept.rst:139]:
```suggestion
Each of the access modes in the table above has an associated tag,
```

File: source/iface/accessor-concept.rst [hunk 3]
```diff
@@ -2,41 +2,400 @@
...
+* If ``DataT`` is not an implicit-lifetime type, the accessor
+  merely allocates uninitialized memory, and the application
+  is responsible for constructing objects in that memory
+  (e.g. by calling placement-new). The application must create an
+  object in each element of the accessor unless the corresponding
+  element of the underlying buffer did not previously contain an object.
+
+If this is a :ref:`ranged accessor <ranged-accessors>`, this applies
+only to the elements within the accessor's range. The content of objects
+in the buffer outside of this range is preserved.
```
Comment (mkinsner) [inline source/iface/accessor-concept.rst:227]:
```suggestion
in the buffer outside of this range are preserved.
```

File: source/iface/accessor-concept.rst (hunk 1 above)
Comment (PetroBondar) [reply source/iface/accessor-concept.rst]: Fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/96/commits/7544af73fd933d50fd209c31178a4e59475a7c93.

File: source/iface/accessor-concept.rst (hunk 2 above)
Comment (PetroBondar) [reply source/iface/accessor-concept.rst]: Fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/96/commits/7544af73fd933d50fd209c31178a4e59475a7c93.

Comment (PetroBondar) [reply source/iface/accessor-concept.rst]: Fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/96/commits/7544af73fd933d50fd209c31178a4e59475a7c93.

**Review by mkinsner — APPROVED** [review]

**Pull request merged**